### PR TITLE
Bug 1416820 — Add a close button to page menu and long press tabs tray menu.

### PR DESCRIPTION
**Pull request by jhugman**
This PR adds a close menu option to the page action menu and the long press tab tray menu.

A followup bug is needed for tests, as we'd like to get this on to the v10.x branch soon.

There is a new string, although this could easily be recycled.

https://bugzilla.mozilla.org/show_bug.cgi?id=1416820

### CODE REVIEW

File: Client/Frontend/Strings.swift [hunk 1]
```diff
@@ -406,6 +406,7 @@ extension Strings {
     public static let AppMenuAddToReadingListTitleString = NSLocalizedString("Menu.AddToReadingList.Title", tableName: "Menu", value: "Add to Reading List", comment: "Label for the button, displayed in the menu, used to add a page to the reading list.")
     public static let AppMenuShowTabsTitleString = NSLocalizedString("Menu.ShowTabs.Title", tableName: "Menu", value: "Show Tabs", comment: "Label for the button, displayed in the menu, used to open the tabs tray")
     public static let AppMenuSharePageTitleString = NSLocalizedString("Menu.SharePageAction.Title", tableName: "Menu", value: "Share Page With…", comment: "Label for the button, displayed in the menu, used to open the share dialog.")
+    public static let AppMenuCloseTabTitleString = NSLocalizedString("Menu.CloseTabAction.Title", tableName: "Menu", value: "Close Tab", comment: "Label for the button, displayed in the menu, used to close the current tab.")
```
Comment (jhugman) [inline Client/Frontend/Strings.swift]: This could be replaced by already existing `Strings.CloseTabTitle`.

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift [hunk 2]
```diff
@@ -200,6 +200,10 @@ extension PhotonActionSheetProtocol {
             presentShareMenu(url, tab, buttonView, .up)
         }
 
+        let closeTab = PhotonActionSheetItem(title: Strings.AppMenuCloseTabTitleString, iconString: "action_remove") { action in
```
Comment (farhanpatel) [inline Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: I'm of the opinion that this shouldnt be in the page options menu. We've got to be careful about what we add here because we will start scrolling on smaller devices.

File: Client/Frontend/Strings.swift (hunk 1 above)
Comment (farhanpatel) [reply Client/Frontend/Strings.swift]: I think it'd be simpler for l10n to reuse the string so we can land this for 10.3

**Comment by tecgirl** [comment]
I think it’s good in both places.

Tl;dr

I also want to mind the amount of items in that list but I think this is worth it. As far as discoverabilty of the LP options, once discovered is a totally time saver. Perhaps we could help spread the word about it. And, as a general good UX practice, particular redundancies are ftw.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -1589,6 +1589,11 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
         controller.addAction(UIAlertAction(title: Strings.NewPrivateTabTitle, style: .default, handler: { _ in
             self.tabManager.addTabAndSelect(isPrivate: true)
         }))
+        controller.addAction(UIAlertAction(title: Strings.CloseTabTitle, style: .default, handler: { _ in
```
Comment (farhanpatel) [inline Client/Frontend/Browser/BrowserViewController.swift]: I think the style for this should be .destructive because this is a delete action

File: Client/Frontend/Strings.swift (hunk 1 above)
Comment (jhugman) [reply Client/Frontend/Strings.swift]: :+1:

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift (hunk 2 above)
Comment (jhugman) [reply Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: I've left this as is, as per @tecgirl 's decision.

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 3 above)
Comment (jhugman) [reply Client/Frontend/Browser/BrowserViewController.swift]: :+1: 

I ummed and ahhed about using `.destructive`. 

Reasons against: "Close All Tabs" isn't `.destructive`; now adding a third color to our menus: black, blue and red.

**Comment by bryanbell** [comment]
Closing tabs shouldn't be an option in the Page Action Menu.
